pkg/rbac/rbac.go

```diff
@@ -0,0 +1,287 @@
+package rbac
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"golang.org/x/exp/slices"
+
+	authorizationv1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+)
+
+// ACLConfig structure captures the configuration related to acl needed to  perform an action
+// e.g. to view a managedcluster, user need the "get" permission on the  managedcluster resource
+// contains ApiGroup, Resource type but  not Version as rules in ClusterRole do not include/specify version
+type ACLConfig struct {
+	gr   schema.GroupResource
+	verb string
+}
+
+// MetricsACLConfig is an instance of ACLConfig containing config for  Metrics Access
+var MetricsACLConfig = ACLConfig{
+	gr: schema.GroupResource{
+		Group:    "cluster.open-cluster-management.io",
+		Resource: "managedclusters",
+	},
+	verb: "metrics/",
+}
+
+// APIEngine is the rbac api for custom/finer grained RBAC, it should preferably be instantiated through
```

Comment:
Since the struct fields are private.
```suggestion
// APIEngine is the rbac api for custom/finer grained RBAC, it must be instantiated through
```